extension/chrome/settings/modules/my_key_update.ts

```diff
@@ -101,45 +101,56 @@ View.run(
+        const updatedKeyPassphrase = String($('.input_passphrase').val());
+        if (typeof updatedKey === 'undefined') {
+          await Ui.modal.warning(Lang.setup.keyFormattedWell(this.prvHeaders.begin, String(this.prvHeaders.end)), Ui.testCompatibilityLink);
+        } else if (updatedKey.isPublic) {
           await Ui.modal.warning(
-            `Key update: This looks like a valid key but it cannot be used for encryption. Please ${Lang.general.contactMinimalSubsentence(
-              !!this.fesUrl
-            )} to see why is that.`,
-            Ui.testCompatibilityLink
+            'This was a public key. Please insert a private key instead. It\'s a block of text starting with "' + this.prvHeaders.begin + '"'
           );
-          window.location.href = this.showKeyUrl;
+          /* eslint-disable @typescript-eslint/no-non-null-assertion */
+        } else if (updatedKey.id !== (await KeyUtil.parse(this.ki!.public)).id) {
+          await Ui.modal.warning(`This key ${Str.spaced(updatedKey.id || 'err')} does not match your current key ${Str.spaced(this.ki!.fingerprints[0])}`);
+          /* eslint-enable @typescript-eslint/no-non-null-assertion */
+        } else if ((await KeyUtil.decrypt(updatedKey, updatedKeyPassphrase)) !== true) {
+          await Ui.modal.error('The pass phrase does not match.\n\nPlease enter pass phrase of the newly updated key.');
+        } else {
+          if (updatedKey.usableForEncryption) {
+            await this.storeUpdatedKeyAndPassphrase(updatedKeyEncrypted, updatedKeyPassphrase);
+            return;
+          }
+          // cannot get a valid encryption key packet
+          if ((await KeyUtil.isWithoutSelfCertifications(updatedKey)) || updatedKey.usableForEncryptionButExpired) {
+            // known issues - key can be fixed
+            const fixedEncryptedPrv = await Settings.renderPrvCompatFixUiAndWaitTilSubmittedByUser(
+              this.acctEmail,
+              '.compatibility_fix_container',
+              updatedKeyEncrypted,
+              updatedKeyPassphrase,
+              this.showKeyUrl
+            );
+            await this.storeUpdatedKeyAndPassphrase(fixedEncryptedPrv, updatedKeyPassphrase);
+          } else {
+            await Ui.modal.warning(
+              `Key update: This looks like a valid key but it cannot be used for encryption. Please ${Lang.general.contactMinimalSubsentence(
+                !!this.fesUrl
+              )} to see why is that.`,
+              Ui.testCompatibilityLink
+            );
+            window.location.href = this.showKeyUrl;
+          }
+        }
+      } catch (e) {
+        if (e instanceof UnexpectedKeyTypeError) {
+          await Ui.modal.warning(`The key provided is invalid. Please ensure that you import an ASCII armored OpenPGP private key and try again.
+          \nFor assistance, please ${Lang.general.contactMinimalSubsentence(!!this.fesUrl)}.`);
```

Comment:
This code will catch only `UnexpectedKeyTypeError` error type and won't send error reports even if other errors will be thrown.

By the way, we already have code for catching errors for invalid keys:
- https://github.com/FlowCrypt/flowcrypt-browser/blob/master/extension/chrome/settings/setup/setup-import-key.ts#L55
- https://github.com/FlowCrypt/flowcrypt-browser/blob/master/extension/chrome/settings/modules/add_key.ts#L118

Let's move it to separate method `handlePrivateKeyError` and use it in all 3 places.

Reply:
@sosnovsky I can see some slight differences between add_key.ts and setup-import-key.ts. I believe it is not difficult but a bit complicated that I thought. Would it be okay if we work on simplification later on the other PR?

Reply:
Which differences do they have? Maybe it can fixed by adding another parameter to `handlePrivateKeyError`?
Will you continue work on this PR or I should check new one https://github.com/FlowCrypt/flowcrypt-browser/pull/5327?

Reply:
> Which differences do they have? Maybe it can fixed by adding another parameter to handlePrivateKeyError?

Some additional steps required in setup-import-key. I will try  to move it in a  separate class and place the handlePrivateKeyError.

> Will you continue work on this PR or I should check new one #5327?

I think I will continue working on the new one https://github.com/FlowCrypt/flowcrypt-browser/pull/5327. I'm currently working on fixing failing test after I have rewritten the flow that catching errors. I hope I can finish it in time.